File: 10/solution.ts
import { readFileSync } from 'fs';

const input = readFileSync('input.txt', 'utf8').split('\n').map(line => [line.split(' ')[0], parseInt(line.split(' ')[1])]) as any[];




const solution1 = (readFileSync('input.txt', 'utf8')
  .split('\n')
  .map(line => [line.split(' ')[0], parseInt(line.split(' ')[1])]) as any[])
  .reduce((acc, line) => (line[0] === 'noop') ? [...acc, acc.at(-1)] : [...acc, acc.at(-1), acc.at(-1) + line[1]], [1]);


const solution2 = input.length;



let solution1_sum = 0
for (let i = 19; i < solution1.length; i += 40) {
  solution1_sum += solution1[i] * (i + 1);
}

console.log(solution1_sum);
console.log(solution2);

File: 10/solution.py

input = []

with open('input.txt', 'r') as f:
  input = f.read().splitlines()
  

solution = [1]

for line in input:
  if(line == 'noop'):
    solution.append(solution[-1])
    
  else:
    amount = int(line.split(' ')[1])
    solution.append(solution[-1])
    solution.append(solution[-1] + amount)
    

  
solution_sum = 0
for i in range(19, len(solution), 40):
    solution_sum += solution[i]* (i+1)
    
    
for i in range(0, len(solution), 40):
  for j in range(i, i+40):
    if(j >= len(solution)-1):
      break
    if(j-i in [x for x in range(solution[j]-1, solution[j]+2)]):
      print('#', end='')
    else:
      print('.', end='')

  print()
    
print(solution_sum)
# print(solution)